WcaOnRails/app/models/competition.rb

```diff
@@ -1763,33 +1763,39 @@ def self.wcif_json_schema
     }
   end
 
+  alias_attribute :venue_address, :venueAddress
+  alias_attribute :venue_details, :venueDetails
+  alias_attribute :short_name, :cellName
+  alias_attribute :city, :cityName
+
+  def country_iso2
+    country&.iso2
+  end
+
+  def url
+    Rails.application.routes.url_helpers.competition_url(self, host: EnvVars.ROOT_URL)
+  end
+
+  DEFAULT_SERIALIZE_OPTIONS = {
+    only: ["id", "name", "website", "start_date", "registration_open",
+           "registration_close", "announced_at", "cancelled_at", "end_date",
+           "competitor_limit"],
+    methods: ["url", "website", "short_name", "city", "venue_address",
+              "venue_details", "latitude_degrees", "longitude_degrees",
+              "country_iso2", "event_ids"],
```

Comment:
Commenting here, but the answer is obviously valid for your other comments: you're not blind, but it's indeed rails' "magic": https://guides.rubyonrails.org/association_basics.html#methods-added-by-has-many

If you have `has_many :events` it will generate the accessors for `event_ids` :)

I didn't comment [here](https://github.com/thewca/worldcubeassociation.org/pull/7163/files#diff-af088cfb59d8f16fcb2a831846d9a2e1c25baa42aa8d00897671bb5f4c6cb60fR31) because I didn't want to make an uneducated guess about a code I didn't fully read, but given your question you may want to check if the method is not already generated for you?
(I couldn't review the full PR nor the context in which it was used, so I may be totally wrong!)

Reply:
Thanks, gotcha! Rails :magic_wand: 

(Indeed in the PR it's setting an `attr_writer` that is deliberately circumventing Rails bindings, different story. If you know how the Delegates / Organisers selectise is handled in the current Competition Form, that one is the same idea.)